README.md
# atom-subnetreplacer package

Converts subnet masks between formats

Example:

Select "/24"

"Convert to netmask" gives:       255.255.255.0  
"Convert to inverse mask" gives:  0.0.0.255  

Select "0.0.0.255"  

"Convert to prefix" gives:        /24  

BETA CODE  
For the moment, no mass conversions (yet)  

![A screenshot of your package](https://f.cloud.github.com/assets/69169/2290250/c35d867a-a017-11e3-86be-cd7c5bf3ff9b.gif)


CHANGELOG.md
## 0.0.1 - First Release
* Very basic
* Only full selection replacement (one-by-one)
* No mass selection changes (yet)
* No key maps (yet)


spec/subnetreplacer-spec.js
'use babel';

import Subnetreplacer from '../lib/subnetreplacer';

// Use the command `window:run-package-specs` (cmd-alt-ctrl-p) to run specs.
//
// To run a specific `it` or `describe` block add an `f` to the front (e.g. `fit`
// or `fdescribe`). Remove the `f` to unfocus the block.

describe('Subnetreplacer', () => {
  let workspaceElement, activationPromise;

  beforeEach(() => {
    workspaceElement = atom.views.getView(atom.workspace);
    activationPromise = atom.packages.activatePackage('subnetreplacer');
  });

  describe('when the subnetreplacer:toggle event is triggered', () => {
    it('hides and shows the modal panel', () => {
      // Before the activation event the view is not on the DOM, and no panel
      // has been created
      expect(workspaceElement.querySelector('.subnetreplacer')).not.toExist();

      // This is an activation event, triggering it will cause the package to be
      // activated.
      atom.commands.dispatch(workspaceElement, 'subnetreplacer:toggle');

      waitsForPromise(() => {
        return activationPromise;
      });

      runs(() => {
        expect(workspaceElement.querySelector('.subnetreplacer')).toExist();

        let subnetreplacerElement = workspaceElement.querySelector('.subnetreplacer');
        expect(subnetreplacerElement).toExist();

        let subnetreplacerPanel = atom.workspace.panelForItem(subnetreplacerElement);
        expect(subnetreplacerPanel.isVisible()).toBe(true);
        atom.commands.dispatch(workspaceElement, 'subnetreplacer:toggle');
        expect(subnetreplacerPanel.isVisible()).toBe(false);
      });
    });

    it('hides and shows the view', () => {
      // This test shows you an integration test testing at the view level.

      // Attaching the workspaceElement to the DOM is required to allow the
      // `toBeVisible()` matchers to work. Anything testing visibility or focus
      // requires that the workspaceElement is on the DOM. Tests that attach the
      // workspaceElement to the DOM are generally slower than those off DOM.
      jasmine.attachToDOM(workspaceElement);

      expect(workspaceElement.querySelector('.subnetreplacer')).not.toExist();

      // This is an activation event, triggering it causes the package to be
      // activated.
      atom.commands.dispatch(workspaceElement, 'subnetreplacer:toggle');

      waitsForPromise(() => {
        return activationPromise;
      });

      runs(() => {
        // Now we can test for view visibility
        let subnetreplacerElement = workspaceElement.querySelector('.subnetreplacer');
        expect(subnetreplacerElement).toBeVisible();
        atom.commands.dispatch(workspaceElement, 'subnetreplacer:toggle');
        expect(subnetreplacerElement).not.toBeVisible();
      });
    });
  });
});


lib/subnetreplacer.js
'use babel';

import { CompositeDisposable } from 'atom';

export default {

  subscriptions: null,

  activate(state) {

    // Events subscribed to in atom's system can be easily cleaned up with a CompositeDisposable
    this.subscriptions = new CompositeDisposable();

    // Register command that toggles this view
    this.subscriptions.add(atom.commands.add('atom-workspace', {'subnetreplacer:tomask': () => this.tomask()}));
    this.subscriptions.add(atom.commands.add('atom-workspace', {'subnetreplacer:toinversemask': () => this.toinversemask()}));
    this.subscriptions.add(atom.commands.add('atom-workspace', {'subnetreplacer:toprefix': () => this.toprefix()}));
  },

  deactivate() {
    this.subscriptions.dispose();
  },


  tomask() {
    const editor = atom.workspace.getActiveTextEditor()
    //var iteratortest = new Function('a','return a.replace("koekoek")');
    if (editor) {
      const selection = editor.getSelectedText()
  //    const selrange = editor.getSelectedBufferRange()
  //    editor.scanInBufferRange('\/\d\d', selrange , iteratortest)
      if (selection == '/32') { editor.insertText(`255.255.255.255`); }
      if (selection == '/31') { editor.insertText(`255.255.255.254`); }
      if (selection == '/30') { editor.insertText(`255.255.255.252`); }
      if (selection == '/29') { editor.insertText(`255.255.255.248`); }
      if (selection == '/28') { editor.insertText(`255.255.255.240`); }
      if (selection == '/27') { editor.insertText(`255.255.255.224`); }
      if (selection == '/26') { editor.insertText(`255.255.255.192`); }
      if (selection == '/25') { editor.insertText(`255.255.255.128`); }
      if (selection == '/24') { editor.insertText(`255.255.255.0`); }
      if (selection == '/23') { editor.insertText('255.255.254.0'); }
      if (selection == '/22') { editor.insertText(`255.255.252.0`); }
      if (selection == '/21') { editor.insertText('255.255.248.0'); }
      if (selection == '/20') { editor.insertText(`255.255.240.0`); }
      if (selection == '/19') { editor.insertText(`255.255.224.0`); }
      if (selection == '/18') { editor.insertText(`255.255.192.0`); }
      if (selection == '/17') { editor.insertText(`255.255.128.0`); }
      if (selection == '/16') { editor.insertText(`255.255.0.0`); }
      if (selection == '/15') { editor.insertText(`255.254.0.0`); }
      if (selection == '/14') { editor.insertText(`255.252.0.0`); }
      if (selection == '/13') { editor.insertText(`255.248.0.0`); }

      if (selection == '0.0.0.0') { editor.insertText(`255.255.255.255`); }
      if (selection == '0.0.0.1') { editor.insertText(`255.255.255.254`); }
      if (selection == '0.0.0.3') { editor.insertText(`255.255.255.252`); }
      if (selection == '0.0.0.7') { editor.insertText(`255.255.255.248`); }
      if (selection == '0.0.0.15') { editor.insertText(`255.255.255.240`); }
      if (selection == '0.0.0.31') { editor.insertText(`255.255.255.224`); }
      if (selection == '0.0.0.63') { editor.insertText(`255.255.255.192`); }
      if (selection == '0.0.0.127') { editor.insertText(`255.255.255.128`); }
      if (selection == '0.0.0.255') { editor.insertText(`255.255.255.0`); }
      if (selection == '0.0.1.255') { editor.insertText('255.255.254.0'); }
      if (selection == '0.0.3.255') { editor.insertText(`255.255.252.0`); }
      if (selection == '0.0.7.255') { editor.insertText('255.255.248.0'); }
      if (selection == '0.0.15.255') { editor.insertText(`255.255.240.0`); }
      if (selection == '0.0.31.255') { editor.insertText(`255.255.224.0`); }
      if (selection == '0.0.63.255') { editor.insertText(`255.255.192.0`); }
      if (selection == '0.0.127.255') { editor.insertText(`255.255.128.0`); }
      if (selection == '0.0.255.255') { editor.insertText(`255.255.0.0`); }
      if (selection == '0.1.255.255') { editor.insertText(`255.254.0.0`); }
      if (selection == '0.3.255.255') { editor.insertText(`255.252.0.0`); }
      if (selection == '0.7.255.255') { editor.insertText(`255.248.0.0`); }
    }
  },

  toinversemask() {
    const editor = atom.workspace.getActiveTextEditor()
    if (editor) {
      const selection = editor.getSelectedText()
      if (selection == '/32') { editor.insertText(`0.0.0.0`); }
      if (selection == '/31') { editor.insertText(`0.0.0.1`); }
      if (selection == '/30') { editor.insertText(`0.0.0.3`); }
      if (selection == '/29') { editor.insertText(`0.0.0.7`); }
      if (selection == '/28') { editor.insertText(`0.0.0.15`); }
      if (selection == '/27') { editor.insertText(`0.0.0.31`); }
      if (selection == '/26') { editor.insertText(`0.0.0.63`); }
      if (selection == '/25') { editor.insertText(`0.0.0.127`); }
      if (selection == '/24') { editor.insertText(`0.0.0.255`); }
      if (selection == '/23') { editor.insertText(`0.0.1.255`); }
      if (selection == '/22') { editor.insertText(`0.0.3.255`); }
      if (selection == '/21') { editor.insertText(`0.0.7.255`); }
      if (selection == '/20') { editor.insertText(`0.0.15.255`); }
      if (selection == '/19') { editor.insertText(`0.0.31.255`); }
      if (selection == '/18') { editor.insertText(`0.0.63.255`); }
      if (selection == '/17') { editor.insertText(`0.0.127.255`); }
      if (selection == '/16') { editor.insertText(`0.0.255.255`); }
      if (selection == '/15') { editor.insertText(`0.1.255.255`); }
      if (selection == '/14') { editor.insertText(`0.3.255.255`); }
      if (selection == '/13') { editor.insertText(`0.7.255.255`); }

      if (selection == '255.255.255.255') { editor.insertText(`0.0.0.0`); }
      if (selection == '255.255.255.254') { editor.insertText(`0.0.0.1`); }
      if (selection == '255.255.255.252') { editor.insertText(`0.0.0.3`); }
      if (selection == '255.255.255.248') { editor.insertText(`0.0.0.7`); }
      if (selection == '255.255.255.240') { editor.insertText(`0.0.0.15`); }
      if (selection == '255.255.255.224') { editor.insertText(`0.0.0.31`); }
      if (selection == '255.255.255.192') { editor.insertText(`0.0.0.63`); }
      if (selection == '255.255.255.128') { editor.insertText(`0.0.0.127`); }
      if (selection == '255.255.255.0') { editor.insertText(`0.0.0.255`); }
      if (selection == '255.255.254.0') { editor.insertText(`0.0.1.255`); }
      if (selection == '255.255.252.0') { editor.insertText(`0.0.3.255`); }
      if (selection == '255.255.248.0') { editor.insertText(`0.0.7.255`); }
      if (selection == '255.255.240.0') { editor.insertText(`0.0.15.255`); }
      if (selection == '255.255.224.0') { editor.insertText(`0.0.31.255`); }
      if (selection == '255.255.192.0') { editor.insertText(`0.0.63.255`); }
      if (selection == '255.255.128.0') { editor.insertText(`0.0.127.255`); }
      if (selection == '255.255.0.0') { editor.insertText(`0.0.255.255`); }
      if (selection == '255.254.0.0') { editor.insertText(`0.1.255.255`); }
      if (selection == '255.252.0.0') { editor.insertText(`0.3.255.255`); }
      if (selection == '255.248.0.0') { editor.insertText(`0.7.255.255`); }



    }
  },

  toprefix() {
    const editor = atom.workspace.getActiveTextEditor()
    if (editor) {
      const selection = editor.getSelectedText()
      if (selection == '0.0.0.0') { editor.insertText(`/32`); }
      if (selection == '0.0.0.1') { editor.insertText(`/31`); }
      if (selection == '0.0.0.3') { editor.insertText(`/30`); }
      if (selection == '0.0.0.7') { editor.insertText(`/29`); }
      if (selection == '0.0.0.15') { editor.insertText(`/28`); }
      if (selection == '0.0.0.31') { editor.insertText(`/27`); }
      if (selection == '0.0.0.63') { editor.insertText(`/26`); }
      if (selection == '0.0.0.127') { editor.insertText(`/25`); }
      if (selection == '0.0.0.255') { editor.insertText(`/24`); }
      if (selection == '0.0.1.255') { editor.insertText('/23'); }
      if (selection == '0.0.3.255') { editor.insertText(`/22`); }
      if (selection == '0.0.7.255') { editor.insertText('/21'); }
      if (selection == '0.0.15.255') { editor.insertText(`/20`); }
      if (selection == '0.0.31.255') { editor.insertText(`/19`); }
      if (selection == '0.0.63.255') { editor.insertText(`/18`); }
      if (selection == '0.0.127.255') { editor.insertText(`/17`); }
      if (selection == '0.0.255.255') { editor.insertText(`/16`); }
      if (selection == '0.1.255.255') { editor.insertText(`/15`); }
      if (selection == '0.3.255.255') { editor.insertText(`/14`); }
      if (selection == '0.7.255.255') { editor.insertText(`/13`); }

      if (selection == '255.255.255.255') { editor.insertText(`/32`); }
      if (selection == '255.255.255.254') { editor.insertText(`/31`); }
      if (selection == '255.255.255.252') { editor.insertText(`/30`); }
      if (selection == '255.255.255.248') { editor.insertText(`/29`); }
      if (selection == '255.255.255.240') { editor.insertText(`/28`); }
      if (selection == '255.255.255.224') { editor.insertText(`/27`); }
      if (selection == '255.255.255.192') { editor.insertText(`/26`); }
      if (selection == '255.255.255.128') { editor.insertText(`/25`); }
      if (selection == '255.255.255.0') { editor.insertText(`/24`); }
      if (selection == '255.255.254.0') { editor.insertText(`/23`); }
      if (selection == '255.255.252.0') { editor.insertText(`/22`); }
      if (selection == '255.255.248.0') { editor.insertText(`/21`); }
      if (selection == '255.255.240.0') { editor.insertText(`/20`); }
      if (selection == '255.255.224.0') { editor.insertText(`/19`); }
      if (selection == '255.255.192.0') { editor.insertText(`/18`); }
      if (selection == '255.255.128.0') { editor.insertText(`/17`); }
      if (selection == '255.255.0.0') { editor.insertText(`/16`); }
      if (selection == '255.254.0.0') { editor.insertText(`/15`); }
      if (selection == '255.252.0.0') { editor.insertText(`/14`); }
      if (selection == '255.248.0.0') { editor.insertText(`/13`); }

    }
  }

};


lib/atom-subnetreplacer.js
'use babel';

import { CompositeDisposable } from 'atom';

export default {

  subscriptions: null,

  activate(state) {

    // Events subscribed to in atom's system can be easily cleaned up with a CompositeDisposable
    this.subscriptions = new CompositeDisposable();

    // Register command that toggles this view
    this.subscriptions.add(atom.commands.add('atom-workspace', {'atom-subnetreplacer:tomask': () => this.tomask()}));
    this.subscriptions.add(atom.commands.add('atom-workspace', {'atom-subnetreplacer:toinversemask': () => this.toinversemask()}));
    this.subscriptions.add(atom.commands.add('atom-workspace', {'atom-subnetreplacer:toprefix': () => this.toprefix()}));
  },

  deactivate() {
    this.subscriptions.dispose();
  },


  tomask() {
    const editor = atom.workspace.getActiveTextEditor()
    //var iteratortest = new Function('a','return a.replace("koekoek")');
    if (editor) {
      const selection = editor.getSelectedText()
  //    const selrange = editor.getSelectedBufferRange()
  //    editor.scanInBufferRange('\/\d\d', selrange , iteratortest)
      if (selection == '/32') { editor.insertText(`255.255.255.255`); }
      if (selection == '/31') { editor.insertText(`255.255.255.254`); }
      if (selection == '/30') { editor.insertText(`255.255.255.252`); }
      if (selection == '/29') { editor.insertText(`255.255.255.248`); }
      if (selection == '/28') { editor.insertText(`255.255.255.240`); }
      if (selection == '/27') { editor.insertText(`255.255.255.224`); }
      if (selection == '/26') { editor.insertText(`255.255.255.192`); }
      if (selection == '/25') { editor.insertText(`255.255.255.128`); }
      if (selection == '/24') { editor.insertText(`255.255.255.0`); }
      if (selection == '/23') { editor.insertText('255.255.254.0'); }
      if (selection == '/22') { editor.insertText(`255.255.252.0`); }
      if (selection == '/21') { editor.insertText('255.255.248.0'); }
      if (selection == '/20') { editor.insertText(`255.255.240.0`); }
      if (selection == '/19') { editor.insertText(`255.255.224.0`); }
      if (selection == '/18') { editor.insertText(`255.255.192.0`); }
      if (selection == '/17') { editor.insertText(`255.255.128.0`); }
      if (selection == '/16') { editor.insertText(`255.255.0.0`); }
      if (selection == '/15') { editor.insertText(`255.254.0.0`); }
      if (selection == '/14') { editor.insertText(`255.252.0.0`); }
      if (selection == '/13') { editor.insertText(`255.248.0.0`); }

      if (selection == '0.0.0.0') { editor.insertText(`255.255.255.255`); }
      if (selection == '0.0.0.1') { editor.insertText(`255.255.255.254`); }
      if (selection == '0.0.0.3') { editor.insertText(`255.255.255.252`); }
      if (selection == '0.0.0.7') { editor.insertText(`255.255.255.248`); }
      if (selection == '0.0.0.15') { editor.insertText(`255.255.255.240`); }
      if (selection == '0.0.0.31') { editor.insertText(`255.255.255.224`); }
      if (selection == '0.0.0.63') { editor.insertText(`255.255.255.192`); }
      if (selection == '0.0.0.127') { editor.insertText(`255.255.255.128`); }
      if (selection == '0.0.0.255') { editor.insertText(`255.255.255.0`); }
      if (selection == '0.0.1.255') { editor.insertText('255.255.254.0'); }
      if (selection == '0.0.3.255') { editor.insertText(`255.255.252.0`); }
      if (selection == '0.0.7.255') { editor.insertText('255.255.248.0'); }
      if (selection == '0.0.15.255') { editor.insertText(`255.255.240.0`); }
      if (selection == '0.0.31.255') { editor.insertText(`255.255.224.0`); }
      if (selection == '0.0.63.255') { editor.insertText(`255.255.192.0`); }
      if (selection == '0.0.127.255') { editor.insertText(`255.255.128.0`); }
      if (selection == '0.0.255.255') { editor.insertText(`255.255.0.0`); }
      if (selection == '0.1.255.255') { editor.insertText(`255.254.0.0`); }
      if (selection == '0.3.255.255') { editor.insertText(`255.252.0.0`); }
      if (selection == '0.7.255.255') { editor.insertText(`255.248.0.0`); }
    }
  },

  toinversemask() {
    const editor = atom.workspace.getActiveTextEditor()
    if (editor) {
      const selection = editor.getSelectedText()
      if (selection == '/32') { editor.insertText(`0.0.0.0`); }
      if (selection == '/31') { editor.insertText(`0.0.0.1`); }
      if (selection == '/30') { editor.insertText(`0.0.0.3`); }
      if (selection == '/29') { editor.insertText(`0.0.0.7`); }
      if (selection == '/28') { editor.insertText(`0.0.0.15`); }
      if (selection == '/27') { editor.insertText(`0.0.0.31`); }
      if (selection == '/26') { editor.insertText(`0.0.0.63`); }
      if (selection == '/25') { editor.insertText(`0.0.0.127`); }
      if (selection == '/24') { editor.insertText(`0.0.0.255`); }
      if (selection == '/23') { editor.insertText(`0.0.1.255`); }
      if (selection == '/22') { editor.insertText(`0.0.3.255`); }
      if (selection == '/21') { editor.insertText(`0.0.7.255`); }
      if (selection == '/20') { editor.insertText(`0.0.15.255`); }
      if (selection == '/19') { editor.insertText(`0.0.31.255`); }
      if (selection == '/18') { editor.insertText(`0.0.63.255`); }
      if (selection == '/17') { editor.insertText(`0.0.127.255`); }
      if (selection == '/16') { editor.insertText(`0.0.255.255`); }
      if (selection == '/15') { editor.insertText(`0.1.255.255`); }
      if (selection == '/14') { editor.insertText(`0.3.255.255`); }
      if (selection == '/13') { editor.insertText(`0.7.255.255`); }

      if (selection == '255.255.255.255') { editor.insertText(`0.0.0.0`); }
      if (selection == '255.255.255.254') { editor.insertText(`0.0.0.1`); }
      if (selection == '255.255.255.252') { editor.insertText(`0.0.0.3`); }
      if (selection == '255.255.255.248') { editor.insertText(`0.0.0.7`); }
      if (selection == '255.255.255.240') { editor.insertText(`0.0.0.15`); }
      if (selection == '255.255.255.224') { editor.insertText(`0.0.0.31`); }
      if (selection == '255.255.255.192') { editor.insertText(`0.0.0.63`); }
      if (selection == '255.255.255.128') { editor.insertText(`0.0.0.127`); }
      if (selection == '255.255.255.0') { editor.insertText(`0.0.0.255`); }
      if (selection == '255.255.254.0') { editor.insertText(`0.0.1.255`); }
      if (selection == '255.255.252.0') { editor.insertText(`0.0.3.255`); }
      if (selection == '255.255.248.0') { editor.insertText(`0.0.7.255`); }
      if (selection == '255.255.240.0') { editor.insertText(`0.0.15.255`); }
      if (selection == '255.255.224.0') { editor.insertText(`0.0.31.255`); }
      if (selection == '255.255.192.0') { editor.insertText(`0.0.63.255`); }
      if (selection == '255.255.128.0') { editor.insertText(`0.0.127.255`); }
      if (selection == '255.255.0.0') { editor.insertText(`0.0.255.255`); }
      if (selection == '255.254.0.0') { editor.insertText(`0.1.255.255`); }
      if (selection == '255.252.0.0') { editor.insertText(`0.3.255.255`); }
      if (selection == '255.248.0.0') { editor.insertText(`0.7.255.255`); }



    }
  },

  toprefix() {
    const editor = atom.workspace.getActiveTextEditor()
    if (editor) {
      const selection = editor.getSelectedText()
      if (selection == '0.0.0.0') { editor.insertText(`/32`); }
      if (selection == '0.0.0.1') { editor.insertText(`/31`); }
      if (selection == '0.0.0.3') { editor.insertText(`/30`); }
      if (selection == '0.0.0.7') { editor.insertText(`/29`); }
      if (selection == '0.0.0.15') { editor.insertText(`/28`); }
      if (selection == '0.0.0.31') { editor.insertText(`/27`); }
      if (selection == '0.0.0.63') { editor.insertText(`/26`); }
      if (selection == '0.0.0.127') { editor.insertText(`/25`); }
      if (selection == '0.0.0.255') { editor.insertText(`/24`); }
      if (selection == '0.0.1.255') { editor.insertText('/23'); }
      if (selection == '0.0.3.255') { editor.insertText(`/22`); }
      if (selection == '0.0.7.255') { editor.insertText('/21'); }
      if (selection == '0.0.15.255') { editor.insertText(`/20`); }
      if (selection == '0.0.31.255') { editor.insertText(`/19`); }
      if (selection == '0.0.63.255') { editor.insertText(`/18`); }
      if (selection == '0.0.127.255') { editor.insertText(`/17`); }
      if (selection == '0.0.255.255') { editor.insertText(`/16`); }
      if (selection == '0.1.255.255') { editor.insertText(`/15`); }
      if (selection == '0.3.255.255') { editor.insertText(`/14`); }
      if (selection == '0.7.255.255') { editor.insertText(`/13`); }

      if (selection == '255.255.255.255') { editor.insertText(`/32`); }
      if (selection == '255.255.255.254') { editor.insertText(`/31`); }
      if (selection == '255.255.255.252') { editor.insertText(`/30`); }
      if (selection == '255.255.255.248') { editor.insertText(`/29`); }
      if (selection == '255.255.255.240') { editor.insertText(`/28`); }
      if (selection == '255.255.255.224') { editor.insertText(`/27`); }
      if (selection == '255.255.255.192') { editor.insertText(`/26`); }
      if (selection == '255.255.255.128') { editor.insertText(`/25`); }
      if (selection == '255.255.255.0') { editor.insertText(`/24`); }
      if (selection == '255.255.254.0') { editor.insertText(`/23`); }
      if (selection == '255.255.252.0') { editor.insertText(`/22`); }
      if (selection == '255.255.248.0') { editor.insertText(`/21`); }
      if (selection == '255.255.240.0') { editor.insertText(`/20`); }
      if (selection == '255.255.224.0') { editor.insertText(`/19`); }
      if (selection == '255.255.192.0') { editor.insertText(`/18`); }
      if (selection == '255.255.128.0') { editor.insertText(`/17`); }
      if (selection == '255.255.0.0') { editor.insertText(`/16`); }
      if (selection == '255.254.0.0') { editor.insertText(`/15`); }
      if (selection == '255.252.0.0') { editor.insertText(`/14`); }
      if (selection == '255.248.0.0') { editor.insertText(`/13`); }

    }
  }

};
